### PR TITLE
GraphQLController as a Service

### PR DESCRIPTION
**Pull request by renatomefi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Documented?   | no
| License       | MIT

- I just want to do some code enhancements and this is a step to symfony 4.0 as well
- I need some help on how to fix the tests, what's the best way to do it?
- Obs: I won't do it for GraphiQLController since my plan is to split it

### CODE REVIEW

**Comment by renatomefi** [comment]
@mcg-web could you help me with the tests?

**Comment by mcg-web** [comment]
@renatomefi  It seems like the service i not properly loaded, I'll give look to this Monday morning, this weekend I'm some way busy :dancer:

File: Resources/config/routing/graphql.yml [hunk 1]
```diff
@@ -1,23 +1,23 @@
 overblog_graphql_endpoint:
     path: /
     defaults:
-        _controller: OverblogGraphQLBundle:Graph:endpoint
+        _controller: overblog_graphql.controller.graphql:endpoint
```
Comment (mcg-web) [inline Resources/config/routing/graphql.yml]: Add `Action` suffix the method nameis not complet

File: Resources/config/routing/graphql.yml [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
         _format: "json"
 
 overblog_graphql_batch_endpoint:
     path: /batch
     defaults:
-        _controller: OverblogGraphQLBundle:Graph:batchEndpoint
+        _controller: overblog_graphql.controller.graphql:batchEndpoint
```
Comment (mcg-web) [inline Resources/config/routing/graphql.yml]: same here

File: Resources/config/routing/graphql.yml [hunk 3]
```diff
@@ -1,23 +1,23 @@
...
         _format: "json"
 
 overblog_graphql_multiple_endpoint:
     path: /graphql/{schemaName}
     defaults:
-        _controller: OverblogGraphQLBundle:Graph:endpoint
+        _controller: overblog_graphql.controller.graphql:endpoint
```
Comment (mcg-web) [inline Resources/config/routing/graphql.yml]: same here

File: Resources/config/routing/graphql.yml [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
         _format: "json"
 
 overblog_graphql_batch_multiple_endpoint:
     path: /graphql/{schemaName}/batch
     defaults:
-        _controller: OverblogGraphQLBundle:Graph:batchEndpoint
+        _controller: overblog_graphql.controller.graphql:batchEndpoint
```
Comment (mcg-web) [inline Resources/config/routing/graphql.yml]: same here

File: Resources/config/services.yml [hunk 5]
```diff
@@ -93,3 +93,13 @@ services:
     overblog_graphql.react.promise_adapter:
         class: Overblog\GraphQLBundle\Executor\Promise\Adapter\ReactPromiseAdapter
         public: false
+
+    overblog_graphql.controller.graphql:
+        class: Overblog\GraphQLBundle\Controller\GraphController
+        public: false
```
Comment (mcg-web) [inline Resources/config/services.yml]: Controller services must be public

File: Controller/GraphController.php [hunk 6]
```diff
@@ -11,13 +11,53 @@
...
 namespace Overblog\GraphQLBundle\Controller;
 
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Overblog\GraphQLBundle\Request as GraphQLRequest;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
-class GraphController extends Controller
+final class GraphController
```
Comment (mcg-web) [inline Controller/GraphController.php]: I'm not for controller as final...

**Comment by mcg-web** [comment]
Here the fix:

Controller services must be public:

```yaml
# Resources/config/services.yml
# ...
    overblog_graphql.controller.graphql:
        class: Overblog\GraphQLBundle\Controller\GraphController
        public: true
        arguments:
            - "@overblog_graphql.request_batch_parser"
            - "@overblog_graphql.request_executor"
            - "@overblog_graphql.request_parser"
            - "%overblog_graphql.handle_cors%"
            - "%overblog_graphql.batching_method%"
```

You forgot `Action` prefix to method names:
```yaml
# Resources/config/routing/graphql.yml
overblog_graphql_endpoint:
    path: /
    defaults:
        _controller: overblog_graphql.controller.graphql:endpointAction
        _format: "json"

overblog_graphql_batch_endpoint:
    path: /batch
    defaults:
        _controller: overblog_graphql.controller.graphql:batchEndpointAction
        _format: "json"

overblog_graphql_multiple_endpoint:
    path: /graphql/{schemaName}
    defaults:
        _controller: overblog_graphql.controller.graphql:endpointAction
        _format: "json"

overblog_graphql_batch_multiple_endpoint:
    path: /graphql/{schemaName}/batch
    defaults:
        _controller: overblog_graphql.controller.graphql:batchEndpointAction
        _format: "json"
```

**Comment by mcg-web** [comment]
push wrong button :1st_place_medal:

**Comment by renatomefi** [comment]
@mcg-web thanks a lot, your tips were directly to the point! It's working now! :)

**Comment by mcg-web** [comment]
Nice, don't forget to remove wip label if you want it to be merge :)

**Comment by renatomefi** [comment]
@mcg-web sure, I think that was it :)

**Comment by mcg-web** [comment]
@renatomefi thank you for contribution :+1:

**Pull request merged**